### PR TITLE
Cluster Branch - Implementation of manifold models to reduce dimensionality on data for visualization

### PR DESCRIPTION
**Pull request by LeoVoltolini**
This branch was developed to implement manifold models with a focus on visualizing potential clusters by reducing the dimensionality of data from the original feature space to a 2D or 3D space. This implementation follows the original BibMon structure, creating manifold models using sklearn.manifold package, in the same manner as sklearn.regressor previously implemented in BibMon library. The manifold models apply distinct functions for fitting and visualizing the data and take advantage of all inherited features implemented in the Generic Model structure.

### CODE REVIEW

**Comment by afraniomelo** [comment]
Hello, @LeoVoltolini.

Thanks for your valuable contribution!!

It was merged after a minor adjustment in the unit tests.

Best regards.
